### PR TITLE
Add containerized workload part 1

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
Add containerized workload

### CODE REVIEW

File: api/v1alpha1/containerized_types.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	// Resources managed by this workload.
+	Resources []cpv1alpha1.TypedReference `json:"resources,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// Containerized is the Schema for the containerizeds API
+// +kubebuilder:resource:categories={oam}
+// +kubebuilder:subresource:status
+type Containerized struct {
```
Comment (wonderflow) [inline api/v1alpha1/containerized_types.go:52]: We need to consider give it a new name. What about:

- `Container` workload
- `Pod` workload
- `VelaCore` workload
- `Vela` workload
- `Core` workload

**Comment by zzxwill** [comment]
@ryanzhang-oss Is this workload one of those three workloads to be created for vela? What's the major difference with ContainerizedWorkload? I generated the CRD, compared with `core.oam.dev_containerizedworkloads.yaml` of oam-kubernetes-runtime and noticed some differences:)

File: api/v1alpha1/containerized_types.go (hunk 1 above)
Comment (ryanzhang-oss) [reply api/v1alpha1/containerized_types.go:52]: @resouer suggested this name and I don't have a strong opinion. I can't seem to think of a better name (including the ones you suggested)

File: pkg/builtin/workloaddefinition/backend.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  childResourceKinds:
+    - apiVersion: apps/v1
+      kind: Deployment
+    - apiVersion: v1
+      kind: Service
+  extension:
+	defaultTraits:
+        - scale
+        - route
+        - monitor
```
Comment (wonderflow) [inline pkg/builtin/workloaddefinition/backend.go:22]: These three CUE files maybe won't work, but we can fix it in following PRs.

**Review by wonderflow — APPROVED** [review]
LGTM, feel free to merge if you feel OK @ryanzhang-oss

**Pull request merged**